File: android/app/src/main/java/com/colageclub/colage/core/design/StateViews.kt
package com.colageclub.colage.core.design

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

// MARK: - Loading State

@Composable
fun LoadingStateView(
    message: String = "Loading...",
    modifier: Modifier = Modifier
) {
    Box(
        modifier = modifier
            .fillMaxSize()
            .background(ColageColors.Background),
        contentAlignment = Alignment.Center
    ) {
        Column(horizontalAlignment = Alignment.CenterHorizontally) {
            CircularProgressIndicator(
                color = LocalThemeColor.current,
                modifier = Modifier.size(36.dp),
                strokeWidth = 3.dp
            )
            Spacer(Modifier.height(16.dp))
            Text(message, style = ColageFonts.Body.copy(color = ColageColors.TextSecondary))
        }
    }
}

// MARK: - Error State with Retry

@Composable
fun ErrorStateView(
    title: String,
    message: String,
    onRetry: (() -> Unit)? = null,
    modifier: Modifier = Modifier
) {
    Box(
        modifier = modifier
            .fillMaxSize()
            .background(ColageColors.Background),
        contentAlignment = Alignment.Center
    ) {
        Column(
            horizontalAlignment = Alignment.CenterHorizontally,
            modifier = Modifier.padding(horizontal = 40.dp)
        ) {
            Icon(
                Icons.Default.WifiOff,
                contentDescription = null,
                tint = ColageColors.TextTertiary,
                modifier = Modifier.size(44.dp)
            )
            Spacer(Modifier.height(16.dp))
            Text(title, style = ColageFonts.Title3.copy(color = ColageColors.TextPrimary))
            Spacer(Modifier.height(8.dp))
            Text(
                message,
                style = ColageFonts.Body.copy(color = ColageColors.TextSecondary),
                textAlign = TextAlign.Center
            )
            if (onRetry != null) {
                Spacer(Modifier.height(20.dp))
                Button(
                    onClick = onRetry,
                    shape = RoundedCornerShape(50),
                    colors = ButtonDefaults.buttonColors(containerColor = LocalThemeColor.current)
                ) {
                    Icon(Icons.Default.Refresh, null, modifier = Modifier.size(18.dp))
                    Spacer(Modifier.width(8.dp))
                    Text("Try Again", style = ColageFonts.BodyBold)
                }
            }
        }
    }
}

// MARK: - Empty State with Illustration

@Composable
fun EmptyStateView(
    title: String,
    subtitle: String,
    emoji: String? = null,
    icon: @Composable (() -> Unit)? = null,
    modifier: Modifier = Modifier
) {
    Box(
        modifier = modifier
            .fillMaxSize()
            .background(ColageColors.Background),
        contentAlignment = Alignment.Center
    ) {
        Column(
            horizontalAlignment = Alignment.CenterHorizontally,
            modifier = Modifier.padding(horizontal = 40.dp)
        ) {
            if (emoji != null) {
                Text(emoji, fontSize = 56.sp)
            } else {
                icon?.invoke() ?: Icon(
                    Icons.Default.PersonOff,
                    contentDescription = null,
                    tint = LocalThemeColor.current.copy(alpha = 0.6f),
                    modifier = Modifier.size(44.dp)
                )
            }
            Spacer(Modifier.height(16.dp))
            Text(title, style = ColageFonts.Title3.copy(color = ColageColors.TextPrimary))
            Spacer(Modifier.height(8.dp))
            Text(
                subtitle,
                style = ColageFonts.Body.copy(color = ColageColors.TextSecondary),
                textAlign = TextAlign.Center
            )
        }
    }
}

// MARK: - Stale Data Banner

@Composable
fun StaleDataBanner(
    lastUpdatedMs: Long,
    modifier: Modifier = Modifier
) {
    val agoText = timeAgoDisplay(lastUpdatedMs)
    Row(
        modifier = modifier
            .background(ColageColors.Warning.copy(alpha = 0.1f), RoundedCornerShape(50))
            .padding(horizontal = 12.dp, vertical = 6.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(6.dp)
    ) {
        Icon(Icons.Default.Schedule, null, tint = ColageColors.Warning, modifier = Modifier.size(12.dp))
        Text(
            "Last updated $agoText",
            style = ColageFonts.Caption.copy(color = ColageColors.Warning)
        )
    }
}

private fun timeAgoDisplay(timestampMs: Long): String {
    val seconds = ((System.currentTimeMillis() - timestampMs) / 1000).toInt()
    return when {
        seconds < 60 -> "just now"
        seconds < 3600 -> "${seconds / 60}m ago"
        else -> "${seconds / 3600}h ago"
    }
}

File: Colage/Core/Design/StateViews.swift
import SwiftUI

// MARK: - Loading State (Skeleton)

struct LoadingStateView: View {
    @Environment(\.themeColor) private var themeColor
    var message: String = "Loading..."

    var body: some View {
        VStack(spacing: 16) {
            ProgressView()
                .tint(themeColor)
                .scaleEffect(1.2)
            Text(message)
                .font(ColageFonts.body)
                .foregroundStyle(ColageColors.textSecondary)
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity)
        .background(ColageColors.background)
    }
}

// MARK: - Error State with Retry

struct ErrorStateView: View {
    @Environment(\.themeColor) private var themeColor
    let title: String
    let message: String
    var retryAction: (() -> Void)? = nil

    var body: some View {
        VStack(spacing: 16) {
            Image(systemName: "wifi.exclamationmark")
                .font(.system(size: 44))
                .foregroundStyle(ColageColors.textTertiary)

            Text(title)
                .font(ColageFonts.title3)
                .foregroundStyle(ColageColors.textPrimary)

            Text(message)
                .font(ColageFonts.body)
                .foregroundStyle(ColageColors.textSecondary)
                .multilineTextAlignment(.center)
                .padding(.horizontal, 40)

            if let retry = retryAction {
                Button(action: retry) {
                    HStack(spacing: 8) {
                        Image(systemName: "arrow.clockwise")
                        Text("Try Again")
                    }
                    .font(ColageFonts.bodyBold)
                    .foregroundStyle(.white)
                    .padding(.horizontal, 24)
                    .padding(.vertical, 12)
                    .background(themeColor)
                    .clipShape(Capsule())
                }
                .padding(.top, 8)
            }
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity)
        .background(ColageColors.background)
    }
}

// MARK: - Empty State with Illustration

struct EmptyStateView: View {
    @Environment(\.themeColor) private var themeColor
    let icon: String
    let title: String
    let subtitle: String
    var emoji: String? = nil

    var body: some View {
        VStack(spacing: 16) {
            if let emoji = emoji {
                Text(emoji)
                    .font(.system(size: 56))
            } else {
                Image(systemName: icon)
                    .font(.system(size: 44))
                    .foregroundStyle(themeColor.opacity(0.6))
            }

            Text(title)
                .font(ColageFonts.title3)
                .foregroundStyle(ColageColors.textPrimary)

            Text(subtitle)
                .font(ColageFonts.body)
                .foregroundStyle(ColageColors.textSecondary)
                .multilineTextAlignment(.center)
                .padding(.horizontal, 40)
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity)
        .background(ColageColors.background)
    }
}

// MARK: - Stale Data Banner

struct StaleDataBanner: View {
    let lastUpdated: Date

    var body: some View {
        HStack(spacing: 6) {
            Image(systemName: "clock.arrow.circlepath")
                .font(.system(size: 12))
            Text("Last updated \(lastUpdated.timeAgoDisplay)")
                .font(ColageFonts.caption)
        }
        .foregroundStyle(ColageColors.warning)
        .padding(.horizontal, 12)
        .padding(.vertical, 6)
        .background(ColageColors.warning.opacity(0.1))
        .clipShape(Capsule())
    }
}

extension Date {
    var timeAgoDisplay: String {
        let seconds = Int(-timeIntervalSinceNow)
        if seconds < 60 { return "just now" }
        if seconds < 3600 { return "\(seconds / 60)m ago" }
        return "\(seconds / 3600)h ago"
    }
}
